commands/user_e2e_test.go

```diff
@@ -136,3 +136,59 @@ func (s *MmctlE2ETestSuite) TestSearchUserCmd() {
 		s.Equal("Unable to find user '"+emailArg+"'", printer.GetErrorLines()[0])
 	})
 }
+
+func (s *MmctlE2ETestSuite) TestDeleteAllUserCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	s.Run("Delete all user  as unpriviliged user", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		confirm := true
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		err := deleteAllUsersCmdF(s.th.Client, cmd, []string{})
+		s.Require().NotNil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 0)
+
+		// expect users not deleted
+		_, err = s.th.App.GetUser(s.th.BasicUser2.Id)
+		s.Require().Nil(err)
+	})
+
+	s.Run("Delete all user as system admin", func() {
+		printer.Clean()
+
+		cmd := &cobra.Command{}
+		confirm := true
+		cmd.Flags().BoolVar(&confirm, "confirm", confirm, "confirm")
+
+		err := deleteAllUsersCmdF(s.th.SystemAdminClient, cmd, []string{})
+		s.Require().NotNil(err)
+		s.Len(printer.GetLines(), 0)
+		s.Len(printer.GetErrorLines(), 0)
+
+		// expect users not deleted
```

Comment:
I would expect to users to be deleted? Could this be related to `config.ServiceSettings.EnableAPIUserDeletion`?

Reply:
That's the strange part, I tried enabling it but still results in 404 when using system admin client, but the endpoint works only for local client. Not sure if the intended flow should be like that too

Reply:
Bear in mind that [this endpoint only works through local mode](https://api.mattermost.com/#tag/users/paths/~1users/delete) by design :slightly_smiling_face:

Reply:
Oh I see, then the test should be okay
thanks @mgdelacroix for the confirmation